### PR TITLE
fix(ComboBox): Correctly update CommonStates on apply template to fix disabled ComboBox visual state not appearing (for all platforms)

### PR DESCRIPTION
**Pull request by agneszitte-nventive**
GitHub Issue: 
Closes unoplatform/Uno.Material#51

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

When IsEnabled="False", the ComboBox is correctly disabled but the disabled visual state doesn't appear for iOS, Android and WASM.

## What is the new behavior?

When IsEnabled="False", the ComboBox is correctly disabled and the disabled visual state is visible on all platforms.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [X] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [X] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs [hunk 1]
```diff
@@ -142,6 +142,7 @@ protected override void OnLoaded()
 		{
 			base.OnLoaded();
 
+			UpdateCommonStates();
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs:145]: Can you test if it fixes the bug if you put this in `OnApplyTemplate()` instead?

If so it should cover some additional edge cases (eg, ComboBox is both Collapsed and disabled, and becomes Visible after it is already loaded)

**Comment by jeromelaban** [comment]
Thanks Agnes! I'm wondering if we would be able to validate this behavior using a known element's color or property, in a UI Test? This is something that will be useful to validate with @davidjohnoliver's Style PR.

File: src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs (hunk 1 above)
Comment (agneszitte-nventive) [reply src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs:145]: Yes, I will test it tomorrow morning. I thought of OnApplyTemplate first but saw the UpdateDropDownState in the OnLoaded and continue to test at this level only after. But yes I will let you know the outcome after my test.

**Comment by agneszitte-nventive** [comment]
Thanks for the info @jeromelaban, I'm sure we can figure out something. I will check with @davidjohnoliver tomorrow.

File: src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs (hunk 1 above)
Comment (agneszitte-nventive) [reply src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs]: Works well in `OnApplyTemplate()` after all the tests, I correctly updated the code for it.
Thanks again for all the great feedback @davidjohnoliver :) !

**Comment by agneszitte-nventive** [comment]
@jeromelaban and @davidjohnoliver The code is updated (moved in OnApplyTemplate) and I added all the proper UI Tests (ComboBox disabled by default and toggling to disable another ComboBox with screenshots for the "Enabled" and "Disabled" states).

**Comment by jeromelaban** [comment]
Nice work Agnes !
